### PR TITLE
 Make application tests @testable import ExampleApp

### LINKED ISSUE #24: @testable import ExampleAppBinary doesn't work
I've mentioned this issue on Slack channel, and wanted to try it with AirBnB's custom fork of `buck`.

Still no luck though.

Adding `@testable import ExampleAppBinary` to any of the tests, e.g. to `BuckSampleTests.swift` or `BuckSampleUITests.swift` causes this error:

```
Command failed with exit code 1.
stderr: App/UITests/BuckSampleUITests.swift:10:18: error: no such module 'ExampleAppBinary'
@testable import ExampleAppBinary
                 ^
```

While I can definitely find the `ExampleAppBinary` module with `find buck-out -name "*ExampleApp*"`

```
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary.swiftmodule
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary-Swift.h
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary.o
```

I can't use it in Unit or UI tests though.

The original issue is here: https://github.com/facebook/buck/issues/2074

P.S.
I know it's open source and all that, but I don't have the time and code base knowledge to locate the change that removed this functionality and revert just the bits that make app module imports work again.

Since this is the only up-to-date buck demo of mix-n-match project there is (🥇), I thought raising it here would draw some attention to it.

Also, I though most of AirBnB's changes made it to master already... At least I was able to use `import` statements in our mix-n-match project with `modular=True` flag using current `master`.

### PR DESCRIPTION
**Pull request by dfed**
Resolves #24

### CODE REVIEW

File: App/Tests/BuckSampleTests.swift [hunk 1]
```diff
@@ -6,13 +6,13 @@
 //
 
 import XCTest
-@testable import Swift2
+@testable import ExampleApp
 
 final class BuckSampleTests: XCTestCase {
 
-    func testMySwiftClassAdding() {
```
Comment (dfed) [inline App/Tests/BuckSampleTests.swift]: This test was moved into `SomeSwift2Tests`

**Comment by bachand** [comment]
Can we verify that the debugger still works with this change? I know that @shepting made some headway on fixing the debugger by moving code from the library into the binary, so I want to be sure we didn't regress that work. Testing in the VC should be fine.

**Comment by dfed** [comment]
Given that our app is currently using this structure I didn't test the debugger explicitly before merging. But I've now validated that the debugger continues work with this change. I think the key to the debugger working was making the binary include a swift file rather than an objc file.

**Comment by bachand** [comment]
Thanks for checking 🙏 

Makes some sense re: source file👌

**Comment by bachand** [comment]
(As much as this stuff ever makes sense haha 🙃)

**Comment by dfed** [comment]
When we moved to having the app code in the main binary, we were effectively introducing swift into the main binary, which likely had the same effect as having a single dummy Swift source file in the binary. I think. Def a bit of conjecture here 😬

**Comment by bachand** [comment]
I’ll buy it 😉

**Pull request merged**